solver/src/settlement_submission/archerapi.rs

```diff
@@ -0,0 +1,76 @@
+//! https://docs.archerdao.io/for-traders/for-traders/traders
+
+use anyhow::{ensure, Result};
+use reqwest::Client;
+use std::time::SystemTime;
+
+const URL: &str = "https://api.archerdao.io/v1/transaction";
+
+#[derive(Clone)]
+pub struct ArcherApi {
+    client: Client,
+    authorization: String,
+}
+
+impl ArcherApi {
+    pub fn new(authorization: String) -> Self {
+        Self {
+            client: Client::new(),
+            authorization,
+        }
+    }
+
+    /// Submit a signed transaction to the archer network.
+    pub async fn submit_transaction(
+        &self,
+        raw_signed_transaction: &[u8],
+        deadline: SystemTime,
+    ) -> Result<()> {
+        let tx = format!("0x{}", hex::encode(raw_signed_transaction));
+        let deadline = deadline
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_secs()
+            .to_string();
+        let body = serde_json::json!({
```

Comment:
Should we maybe log request and response bodies to make debugging easier in case we run into issues?

Reply:
I would be in favour of logging the request and response together at once but only in the case that the response is an Error.

Reply:
This code includes the response body in the error message. I am hesitant about the request body because it contains the authorization string which should be secret and the full transaction bytes are fairly long and not interesting (or are they?).

Reply:
> This code includes the response body in the error message.

Nice. For some reason I missed that. I think that's fine, so we don't have to log the response every time.

> I am hesitant about the request body because it contains the authorization string

From the code, I am under the impression it is included in the header and not the request body.

> the full transaction bytes are fairly long and not interesting (or are they?).

Hmm, that's a good question. In theory we log this in our transport implementation, so it might help when debugging issues. In my opinion, it would be nice to log the request JSON as `debug` and make this module only print info by default like with the HTTP transport in `shared::arguments::Arguments.log_filter`.

Reply:
You're right the authorization is only in the header. Added the debug log and set default log filter to info.